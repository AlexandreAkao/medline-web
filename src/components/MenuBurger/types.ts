import { Props } from 'react-burger-menu';

type MenuTypes =
  | 'slide'
  | 'stack'
  | 'elastic'
  | 'bubble'
  | 'push'
  | 'pushRotate'
  | 'scaleDown'
  | 'scaleRotate'
  | 'fallDown'
  | 'reveal';

export interface IMenuBurgerProps extends Props {
  type?: MenuTypes;
  children: React.ReactNode;
}
