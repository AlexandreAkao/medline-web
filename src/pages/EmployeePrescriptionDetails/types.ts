export enum UpdateRequestEnum {
  'accept' = 1,
  'refuse' = 2,
}
