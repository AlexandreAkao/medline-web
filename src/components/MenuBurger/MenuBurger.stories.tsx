import { ComponentStory, ComponentMeta } from '@storybook/react';

import MenuBurger from 'components/MenuBurger';

export default {
  title: 'MenuBurger',
  component: MenuBurger,
} as ComponentMeta<typeof MenuBurger>;

const Template: ComponentStory<typeof MenuBurger> = args => <MenuBurger {...args} />;

const options = (
  <div style={{ display: 'flex', flexDirection: 'column' }}>
    <span>Option 1</span>
    <span>Option 2</span>
    <span>Option 3</span>
    <span>Option 4</span>
  </div>
);

export const Slide = Template.bind({});
Slide.args = {
  children: options,
  type: 'slide',
};

export const Bubble = Template.bind({});
Bubble.args = {
  children: options,
  type: 'bubble',
};

export const Elastic = Template.bind({});
Elastic.args = {
  children: options,
  type: 'elastic',
};

export const FallDown = Template.bind({});
FallDown.args = {
  children: options,
  type: 'fallDown',
};

export const Push = Template.bind({});
Push.args = {
  children: options,
  type: 'push',
};

export const PushRotate = Template.bind({});
PushRotate.args = {
  children: options,
  type: 'pushRotate',
};

export const Reveal = Template.bind({});
Reveal.args = {
  children: options,
  type: 'reveal',
};

export const ScaleDown = Template.bind({});
ScaleDown.args = {
  children: options,
  type: 'scaleDown',
};

export const ScaleRotate = Template.bind({});
ScaleRotate.args = {
  children: options,
  type: 'scaleRotate',
};

export const Stack = Template.bind({});
Stack.args = {
  children: options,
  type: 'stack',
};
