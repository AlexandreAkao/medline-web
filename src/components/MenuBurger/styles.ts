import styled from 'styled-components';

import colors from 'styles/colors';

export const MenuBurgerContainer = styled.div`
  .bm-burger-button {
    position: fixed;
    width: 3rem;
    height: 30px;
    right: 3rem;
    top: 3rem;
  }

  .bm-burger-bars {
    background: ${colors.black.soft};
    transition: 0.1s ease all;
  }

  .bm-burger-bars-hover {
    background: ${colors.grey.dark};
    transition: 0.1s ease all;
  }

  .bm-cross-button {
    height: 24px;
    width: 24px;
  }

  .bm-cross {
    background: ${colors.grey.dark};
  }

  .bm-menu-wrap {
    position: fixed;
    height: 100%;
    top: 0;
  }

  .bm-menu {
    background: ${colors.primary.light};
    padding: 2.5em 1.5em 0;
    font-size: 1.15em;
  }

  .bm-morph-shape {
    fill: #373a47;
  }

  .bm-item-list {
    color: #b8b7ad;
    padding: 0.8em;
  }

  .bm-item {
    display: inline-block;
  }

  .bm-overlay {
    background: rgba(0, 0, 0, 0.3);
  }
`;
