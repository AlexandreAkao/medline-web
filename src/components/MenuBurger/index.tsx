import ReactBurgerMenu from 'react-burger-menu';

import { IMenuBurgerProps } from 'components/MenuBurger/types';
import { MenuBurgerContainer } from 'components/MenuBurger/styles';

function MenuBurger({
  type = 'slide',
  right = true,
  noOverlay = true,
  children,
  ...rest
}: IMenuBurgerProps) {
  const Menu = ReactBurgerMenu[type];

  return (
    <MenuBurgerContainer>
      {Menu && (
        <Menu right={right} noOverlay={noOverlay} {...rest}>
          {children}
        </Menu>
      )}
    </MenuBurgerContainer>
  );
}

export default MenuBurger;
