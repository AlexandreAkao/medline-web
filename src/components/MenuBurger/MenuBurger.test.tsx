import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { vi } from 'vitest';

import MenuBurger from 'components/MenuBurger';
import { IMenuBurgerProps } from 'components/MenuBurger/types';

describe('MenuBurger', () => {
  const renderComponent = (customProps: Partial<IMenuBurgerProps> = {}) => {
    const props: IMenuBurgerProps = {
      children: undefined,
      ...customProps,
    };
    return render(<MenuBurger {...props} />);
  };

  it('should match with snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('should open and close MenuBurger', async () => {
    const onOpen = vi.fn();
    const onClose = vi.fn();
    renderComponent({ onOpen, onClose });

    const menu = screen.getByRole('button');
    await userEvent.click(menu);

    const crossClose = screen.getByText('Close Menu');
    await userEvent.click(crossClose);

    expect(onOpen).toBeCalled();
    expect(onClose).toBeCalled();
  });
});
