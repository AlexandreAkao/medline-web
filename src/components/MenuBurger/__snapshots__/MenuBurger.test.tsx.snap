// Vitest Snapshot v1

exports[`MenuBurger > should match with snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ bUMWtq"
  >
    <div>
      <div>
        <div
          class="bm-burger-button"
          style="z-index: 1000;"
        >
          <button
            id="react-burger-menu-btn"
            style="position: absolute; left: 0px; top: 0px; z-index: 1; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
            type="button"
          >
            Open Menu
          </button>
          <span>
            <span
              class="bm-burger-bars"
              style="position: absolute; height: 20%; left: 0px; right: 0px; top: 0%; opacity: 1;"
            />
            <span
              class="bm-burger-bars"
              style="position: absolute; height: 20%; left: 0px; right: 0px; top: 40%; opacity: 1;"
            />
            <span
              class="bm-burger-bars"
              style="position: absolute; height: 20%; left: 0px; right: 0px; top: 80%; opacity: 1;"
            />
          </span>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="bm-menu-wrap"
        id=""
        style="position: fixed; right: 0px; z-index: 1100; width: 300px; height: 100%; transform: translate3d(100%, 0, 0); transition: all 0.5s;"
      >
        <div
          class="bm-menu"
          style="height: 100%; box-sizing: border-box; overflow: auto;"
        >
          <nav
            class="bm-item-list"
            style="height: 100%;"
          />
        </div>
        <div>
          <div
            class="bm-cross-button"
            style="position: absolute; width: 24px; height: 24px; right: 8px; top: 8px;"
          >
            <button
              id="react-burger-cross-btn"
              style="position: absolute; left: 0px; top: 0px; z-index: 1; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
              tabindex="-1"
              type="button"
            >
              Close Menu
            </button>
            <span
              style="position: absolute; top: 6px; right: 14px;"
            >
              <span
                class="bm-cross"
                style="position: absolute; width: 3px; height: 14px; transform: rotate(45deg);"
              />
              <span
                class="bm-cross"
                style="position: absolute; width: 3px; height: 14px; transform: rotate(-45deg);"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
